src/components/mgt-person/mgt-person.ts

```diff
@@ -452,11 +455,21 @@ export class MgtPerson extends MgtTemplatedComponent {
    */
   protected renderImage(personDetails: IDynamicPerson, imageSrc: string) {
     const title =
-      personDetails && this.personCardInteraction === PersonCardInteraction.none ? personDetails.displayName : '';
+      personDetails && this.personCardInteraction === PersonCardInteraction.none
+        ? personDetails.displayName || getEmailFromGraphEntity(personDetails) || ''
+        : '';
+
+    if (imageSrc && !this._isInvalidImageSrc) {
```

Comment:
This will recreate the image on every render. Instead of creating an image, try defining the image in the template

```
<img alt=${title} src=${imageSrc} @error=${e => {}} />
```

Also, consider using @internalProperty for the `this._isInvalidImageSrc` to avoid calling `requestUpdate`. See example: https://github.com/microsoftgraph/microsoft-graph-toolkit/blob/32d7d6a4aac5972d8ae4f560c36df88a75f8432b/src/components/mgt-person/mgt-person.ts#L310

Reply:
Good call. I haven't seen the `@internalProperty` decorator before. I'll have to look it up!